platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -49,7 +49,8 @@ void OfflineDatabase::ensureSchema() {
             case 1: break; // cache-only database; ok to delete
             case 2: migrateToVersion3(); // fall through
             case 3: migrateToVersion4(); // fall through
```

Comment:
I'm fine with that too.

Can you create `MigrateFromV3Schema` and `MigrateFromV4Schema` testcases, using checked-in database fixtures generated by upgrading the existing `test/fixtures/offline_database/v2.db` fixture to v3 and then v4 prior to this PR?